tool_helper_utils/__init__.py

```diff
@@ -2,12 +2,16 @@
 import sys
 
 
+def factory_detector_factory():
```

Comment:
This should really be a class so different callers can easily customize its behavior.

Reply:
Embrace, extend, extinguish. I like it 🎸